src/eradius_server.erl

```diff
@@ -92,22 +93,27 @@
 -callback radius_request(#radius_request{}, #nas_prop{}, HandlerData :: term()) -> 
     {reply, #radius_request{}} | noreply | {error, timeout}.
 
-%% @private
 -spec start_link(atom(), inet:ip4_address(), port_number()) -> {ok, pid()} | {error, term()}.
-start_link(ServerName, IP = {A,B,C,D}, Port) ->
+start_link(ServerName, IP, Port) ->
+    start_link(ServerName, IP, Port, []).
+
+-spec start_link(atom(), inet:ip4_address(), port_number(), [inet:socket_setopt()]) -> {ok, pid()} | {error, term()}.
+start_link(ServerName, IP = {A,B,C,D}, Port, Opts) ->
     Name = list_to_atom(lists:flatten(io_lib:format("eradius_server_~b.~b.~b.~b:~b", [A,B,C,D,Port]))),
-    gen_server:start_link({local, Name}, ?MODULE, {ServerName, IP, Port}, []).
+    gen_server:start_link({local, Name}, ?MODULE, {ServerName, IP, Port, Opts}, []).
 
 stats(Server, Function) ->
     gen_server:call(Server, {stats, Function}).
 
 %% ------------------------------------------------------------------------------------------
 %% -- gen_server Callbacks
 %% @private
-init({ServerName, IP, Port}) ->
+init({ServerName, IP, Port, Opts}) ->
     process_flag(trap_exit, true),
-    RecBuf = application:get_env(eradius, recbuf, 8192),
-    case gen_udp:open(Port, [{active, once}, {ip, IP}, binary, {recbuf, RecBuf}]) of
+    ExtraServerOptions = proplists:get_value(socket_opts, Opts, []),
+    DefaultRecBuf = application:get_env(eradius, recbuf, 8192),
```

Comment:
This is a little bit confusing. As `recbuf` could be passed in `socket_opts` as well. I would suggest to get rid of this one opts and use recbuf only from `socket_opts`. @RoadRunnr any thoughts on this?

Reply:
I added a mechanism to preserve the actual behaviour: if the user doesn't specify nothing in the `socket_opts` we take the default

Reply:
ah, sorry, my fault. Didn't notice that you're getting recbuf from the proplist below

Reply:
I have re-checked server code and it might be that 2 recbuf will be passed in a case when `ExtraServerOptions` from `[{recbuf, RecBuf} | ExtraServerOptions]` already contains `recbuf`. In this case the buffer will be doubled:

```
1> {ok, Sock} = gen_udp:open(5000, [{recbuf, 5001} | [{recbuf, 5001}]]).
{ok,#Port<0.4>}
2> inet:getopts(Sock, [recbuf]).
{ok,[{recbuf,10002}]}
```

which is kind of unexpected behavior I think

Reply:
thanks for this check I'm just testing a fix for this

Reply:
@0xAX done